core/amber/src/test/scala/edu/uci/ics/texera/workflow/common/storage/FileDocumentSpec.scala

```diff
@@ -86,8 +86,8 @@ class FileDocumentSpec extends AnyFlatSpec with Matchers with BeforeAndAfter {
     val futures = (1 to numberOfThreads).map { _ =>
       Future {
         val contentStream = new ByteArrayInputStream(s"Content from thread".getBytes)
-        // multiple document of the same URI try to do write here
-        new FileDocument(tempFileURI).appendStream(contentStream)
+        // Use the same FileDocument instance for all threads
+        fileDocument.appendStream(contentStream)
```

Comment:
if I create another FileDocument instance with the same URI, it will not be thread safe?

Notice that in workers, they will each have an instance of the FileDocument to write to, and they will write to the same URI. 

Thus I don't think the test case is incorrect. The thread-safe should be guaranteed with different FileDocument instances to the same File URI.

Reply:
Currently the `FileDocument` doesn't provide such thread safety over the same URI across multiple `FileDocument` instance. It use `ReentrantReadWriteLock`, which only guarantees that using the same `FileDocument` object will have the writer's thread safety.

So this modification ensures the pass this test under current FileDocument implementation. 

To achieve what you describe, I think a separate PR is needed to change the implementation FileDocument, as well as making the tests "rollback" to current version in the master.

Reply:
ok